homeassistant/components/purpleair/sensor.py

```diff
@@ -0,0 +1,220 @@
+        native_unit_of_measurement=CONCENTRATION_MICROGRAMS_PER_CUBIC_METER,
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda sensor: sensor.pm10_0,
+    ),
+    PurpleAirSensorEntityDescription(
+        key="pm2.5_count_concentration",
+        name="PM2.5 count concentration",
+        entity_registry_enabled_default=False,
+        icon="mdi:blur",
+        native_unit_of_measurement=CONCENTRATION_PARTICLES_PER_100_MILLILITERS,
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda sensor: sensor.pm2_5_um_count,
+    ),
+    PurpleAirSensorEntityDescription(
+        key="pm2.5_mass_concentration",
+        name="PM2.5 mass concentration",
+        device_class=SensorDeviceClass.PM25,
+        native_unit_of_measurement=CONCENTRATION_MICROGRAMS_PER_CUBIC_METER,
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda sensor: sensor.pm2_5,
+    ),
+    PurpleAirSensorEntityDescription(
+        key="pm5.0_count_concentration",
+        name="PM5.0 count concentration",
+        entity_registry_enabled_default=False,
+        icon="mdi:blur",
+        native_unit_of_measurement=CONCENTRATION_PARTICLES_PER_100_MILLILITERS,
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda sensor: sensor.pm5_0_um_count,
+    ),
+    PurpleAirSensorEntityDescription(
+        key="pressure",
+        name="Pressure",
+        device_class=SensorDeviceClass.PRESSURE,
+        native_unit_of_measurement=UnitOfPressure.MBAR,
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda sensor: sensor.pressure,
+    ),
+    PurpleAirSensorEntityDescription(
+        key="rssi",
+        name="RSSI",
+        device_class=SensorDeviceClass.SIGNAL_STRENGTH,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+        native_unit_of_measurement=SIGNAL_STRENGTH_DECIBELS_MILLIWATT,
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda sensor: sensor.pressure,
+    ),
+    PurpleAirSensorEntityDescription(
+        key="temperature",
+        name="Temperature",
+        device_class=SensorDeviceClass.TEMPERATURE,
+        native_unit_of_measurement=UnitOfTemperature.FAHRENHEIT,
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda sensor: sensor.temperature,
+    ),
+    PurpleAirSensorEntityDescription(
+        key="uptime",
+        name="Uptime",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+        icon="mdi:timer",
+        native_unit_of_measurement=UnitOfTime.MINUTES,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        value_fn=lambda sensor: sensor.uptime,
+    ),
+    PurpleAirSensorEntityDescription(
+        key="voc",
+        name="VOC",
+        device_class=SensorDeviceClass.VOLATILE_ORGANIC_COMPOUNDS,
+        native_unit_of_measurement=CONCENTRATION_IAQ,
```

Comment:
The VOC device class should use ug/m3 as unit.

https://developers.home-assistant.io/docs/core/entity/sensor#available-device-classes

Reply:
https://github.com/home-assistant/core/pull/84896